### PR TITLE
testcase/le_tc/filesystem: Add config dependency (CONFIG_FS_SMARTFS)

### PR DESCRIPTION
**Pull request by NilabhShant**
This patch adds config check for smartfs while running the smartfs testcases.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [88769b45e45b5ebe7a08413d982259c453eb0d07] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [88769b45e45b5ebe7a08413d982259c453eb0d07] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 1]
```diff
@@ -3767,7 +3767,7 @@ int tc_filesystem_main(int argc, char *argv[])
 #ifdef CONFIG_ITC_FS
 	itc_fs_main();
 #endif
-#if defined(CONFIG_TC_FS_PROCFS) && !defined(CONFIG_SMARTFS_MULTI_ROOT_DIRS) && !defined(CONFIG_BUILD_PROTECTED)
+#if defined(CONFIG_TC_FS_PROCFS) && !defined(CONFIG_SMARTFS_MULTI_ROOT_DIRS) && !defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_FS_SMARTFS)
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: When you define conditional operation like a if or something else, you should check cheaper first.
Actually CONFIG_SMARTFS_MULTI_ROOT_DIRS is tightly related to CONFIG_FS_SMARTFS

Comment (NilabhShant) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Ok, I fixed it.
Thanks for comment.

**Comment by seinfra** [comment]
Target : [40584d52b920b5c021b0f593b35b7338f4e78323] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [40584d52b920b5c021b0f593b35b7338f4e78323] - Code Rule Check OK.

**Pull request merged**